### PR TITLE
Ionic profile

### PR DESCRIPTION
**Pull request by NicoLorenzo1**
Changes: 
-Profile page on ionic
-Edit button with ion-modal
-Styles css in profile

### CODE REVIEW

**Comment by FrancoBarlocco** [comment]
You should use short comments to explain the features of the profile

File: client/src/app/app.module.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ import { MatTabsModule } from '@angular/material/tabs';
 import { TopNavbarComponent } from './components/top-navbar/top-navbar.component';
 import { MatProgressSpinnerModule } from "@angular/material/progress-spinner";
 import { LoadingComponent } from './components/loading/loading.component';
-
+import { HttpClient } from '@angular/common/http';
```
Comment (alfonsoLeandro) [inline client/src/app/app.module.ts:26]: This is not needed here, you should import HttpClientModule, not HttpClient

File: client/src/app/pages/profile/profile.component.ts [hunk 2]
```diff
@@ -10,18 +11,24 @@ export class ProfileComponent {
 
   publications: Post[] = [];
 
-  constructor(private readonly profileService: ProfileService,) {
-    this.publications = this.profileService.getPublications();
-
+  constructor(private readonly postService: PostsService) {
+    this.postService.getAllPublicationsPerson(9).subscribe({
```
Comment (alfonsoLeandro) [inline client/src/app/pages/profile/profile.component.ts:15]: Add a TODO comment here, so as to not forget to add the actual user ID from user service, instead of the hardcoded "9" ID.

File: patitasuy/src/app/pages/profile/profile.component.scss [hunk 3]
```diff
@@ -0,0 +1,125 @@
+.container-iconPets {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    width: 30px;
```
Comment (alfonsoLeandro) [inline patitasuy/src/app/pages/profile/profile.component.scss]: Use relative units!

File: patitasuy/src/app/pages/profile/profile.component.ts [hunk 4]
```diff
@@ -1,14 +1,84 @@
...
+  name: string = "prueba";
+  email: string = "prueba2";
 
-  constructor() { }
 
-  ngOnInit() {}
+  publications: Post[] = [];
 
+  constructor(private readonly postService: PostsService) {
+    /*
```
Comment (alfonsoLeandro) [inline patitasuy/src/app/pages/profile/profile.component.ts]: Remove commented out code

File: client/src/app/services/profile.service.ts
Comment (alfonsoLeandro) [inline client/src/app/services/profile.service.ts:1]: Delete this file, there's no need for a profileService, as everything in the profile can be done using postService and UserService

File: server/routes/post.routes.ts [hunk 5]
```diff
@@ -1,8 +1,10 @@
 import { Router } from "express";
-import { getPosts } from "../controllers/post.controller";
-
+import { getPosts, getPostsByUser } from "../controllers/post.controller";
 const router = Router();
 
 router.get('/', getPosts);
+router.get('/user/:userId', getPostsByUser);
```
Comment (alfonsoLeandro) [inline server/routes/post.routes.ts]: Nicely done!

File: patitasuy/src/app/services/posts.service.ts [hunk 6]
```diff
@@ -49,7 +51,11 @@ export class PostsService {
         return of(true).pipe(delay(1000));
     }
 
-    markAsFavorite(postId: number, favorite: boolean){
+    markAsFavorite(postId: number, favorite: boolean) {
         return of(this.fakePost).pipe(delay(1000));
     }
+
+    getAllPublicationsPerson(id: number) {
```
Comment (alfonsoLeandro) [inline patitasuy/src/app/services/posts.service.ts]: Name change suggestion: getAllPostsByUser (just a suggestion)

File: patitasuy/src/app/pages/profile/profile.component.html [hunk 7]
```diff
@@ -1,9 +1,63 @@
...
+          </ion-item>
+          <ion-item>
+            <ion-label position="stacked">Ingresa tu email</ion-label>
+            <ion-input type="text" placeholder="Email" [(ngModel)]="email"></ion-input>
+          </ion-item>
+        </ion-content>
+      </ng-template>
+    </ion-modal>
+  </div>
+  <ion-segment (ionChange)="selectPublications($event)">
```
Comment (alfonsoLeandro) [inline patitasuy/src/app/pages/profile/profile.component.html]: Add the attribute "value" with the value "mis-publicaciones" so as to leave the mis-publicaciones tab as selected

File: patitasuy/src/app/pages/profile/profile.component.html [hunk 8]
```diff
@@ -1,9 +1,63 @@
...
+          </ion-toolbar>
+        </ion-header>
+        <ion-content class="ion-padding">
+          <ion-item>
+            <ion-label position="stacked">Ingresa tu nombre</ion-label>
+            <ion-input type="text" placeholder="Nombre" [(ngModel)]="name"></ion-input>
+          </ion-item>
+          <ion-item>
+            <ion-label position="stacked">Ingresa tu email</ion-label>
+            <ion-input type="text" placeholder="Email" [(ngModel)]="email"></ion-input>
```
Comment (alfonsoLeandro) [inline patitasuy/src/app/pages/profile/profile.component.html]: Ionic warns you add a "label" attribute to every ion-input

File: patitasuy/src/app/pages/profile/profile.component.html [hunk 9]
```diff
@@ -1,9 +1,63 @@
...
+  <div class="userContent">
+    <ion-icon class="container-iconAccount" name="person"></ion-icon>
+    <div class="userDates">
+      <p class="userName">{{name}}</p>
+      <p class="userEmail">{{email}}</p>
+    </div>
+    <ion-button id="open-modal" class="editButton" expand="block">
+      <ion-icon name="create-outline"></ion-icon>
+    </ion-button>
+    <ion-modal trigger="open-modal">
```
Comment (alfonsoLeandro) [inline patitasuy/src/app/pages/profile/profile.component.html]: Consider moving this modal to a separate component
pros:
- More organized, cleaner code
- Easier to mantain

cons:
- You will have to use Input() and Output(), which I am not sure you have used before

**Review by alfonsoLeandro — CHANGES_REQUESTED** [review]
Waiting for your thoughts on my coments

File: patitasuy/src/app/pages/profile/profile.component.html (hunk 7 above)
Comment (NicoLorenzo1) [reply patitasuy/src/app/pages/profile/profile.component.html]: I agree with you, its a good change

File: patitasuy/src/app/pages/profile/profile.component.ts (hunk 4 above)
Comment (NicoLorenzo1) [reply patitasuy/src/app/pages/profile/profile.component.ts]: I forgot to remove the comments

File: patitasuy/src/app/pages/profile/profile.component.html (hunk 9 above)
Comment (NicoLorenzo1) [reply patitasuy/src/app/pages/profile/profile.component.html]: Ok its a good change, I implemented it

File: patitasuy/src/app/components/edit-profile/edit-profile.component.ts
Comment (alfonsoLeandro) [inline patitasuy/src/app/components/edit-profile/edit-profile.component.ts:1]: Nicely done!

File: patitasuy/src/app/pages/profile/profile.component.ts [hunk 10]
```diff
@@ -1,14 +1,71 @@
...
+  cancel() {
+    this.modal.dismiss(null, 'cancel');
+  }
+
+  confirm() {
+    this.modal.dismiss(this.name, 'confirm');
+    this.modal.dismiss(this.email, 'confirm');
+  }
+
+  async openModal() {
```
Comment (alfonsoLeandro) [inline patitasuy/src/app/pages/profile/profile.component.ts:55]: Nicely done!

File: patitasuy/src/app/pages/profile/profile.component.ts [hunk 11]
```diff
@@ -1,14 +1,71 @@
...
 
 @Component({
   selector: 'app-profile',
   templateUrl: './profile.component.html',
   styleUrls: ['./profile.component.scss'],
 })
-export class ProfileComponent  implements OnInit {
+export class ProfileComponent {
+  ngOnInit() { }
+  @ViewChild(IonModal) modal!: IonModal;
```
Comment (alfonsoLeandro) [inline patitasuy/src/app/pages/profile/profile.component.ts]: You don't need this don't more

File: server/routes/post.routes.ts [hunk 12]
```diff
@@ -1,9 +1,8 @@
 import { Router } from "express";
-import { getIsFavorite, getPosts } from "../controllers/post.controller";
-
+import { getPosts, getFavoritePosts } from "../controllers/post.controller";
 const router = Router();
 
 router.get('/', getPosts);
-router.get('/isFavorite/post/:postId/user/:userId', getIsFavorite);
```
Comment (alfonsoLeandro) [inline server/routes/post.routes.ts:7]: revert this change

**Review by alfonsoLeandro — APPROVED** [review]
LGMT
